### PR TITLE
Automating Event Logging Infrastructure

### LINKED ISSUE #1953: Enable build time generation of Eventing Logic in coreclr
Right now the eventing functionality for coreclr on both windows and Linux depend on Generated Files which are checked into the repo. These files should be removed from the repo and  must be generated and consumed at build time

### PR DESCRIPTION
**Pull request by ramarag**
With this change the infrastructure required for Event Logging will be generated as part of build
 resolves #1953

### CODE REVIEW

**Comment by ramarag** [comment]
@brianrob @gkhanna79 @schellap  Can you take a look ?

File: build.cmd [hunk 1]
```diff
@@ -197,6 +198,26 @@ exit /b 1
 
 REM Build CoreCLR
 :BuildCoreCLR
+:: Run Steps to Generate ETW specific infrastructure the 
+if not exist "%__GeneratedIntermediatesDir%" md "%__GeneratedIntermediatesDir%"
```
Comment (gkhanna79) [inline build.cmd]: Do you want to delete this folder if it exists to delete any previously created files?

File: build.cmd [hunk 2]
```diff
@@ -197,6 +198,26 @@ exit /b 1
...
+if not exist "%__GeneratedIntermediatesDir%\inc" md "%__GeneratedIntermediatesDir%\inc"
+set "genetw=%__SourceDir%\scripts\genWinEtw.py"
+
+mc -h "%__GeneratedIntermediatesDir%\inc" -r "%__GeneratedIntermediatesDir%" -b -co -um -p FireEtw "%__SourceDir%\VM\ClrEtwAll.man"
+IF ERRORLEVEL 1 goto FailedToGenEtwMetadata 
+
+echo generating clretwallmain.h and etmdummy.h
+"%PythonPath%"  "%genetw%" --man "%__SourceDir%\VM\ClrEtwAll.man" --exc "%__SourceDir%\VM\ClrEtwAllMeta.lst" --eventheader "%__GeneratedIntermediatesDir%\inc\ClrEtwAll.h" --macroheader "%__GeneratedIntermediatesDir%\inc\clretwallmain.h" --dummy "%__GeneratedIntermediatesDir%\inc\etmdummy.h"
```
Comment (gkhanna79) [inline build.cmd]: Can you please check the existence of Python and aborting the script if not found?

**Comment by gkhanna79** [comment]
Aside from few comments, assuming you have tested this, it looks fine to me.

File: build.cmd (hunk 2 above)
Comment (ramarag) [inline build.cmd]: the logic for this check is present in https://github.com/ramarag/coreclr/blob/event_logging_automation/src/pal/tools/probe-win.ps1#L57

File: build.cmd (hunk 1 above)
Comment (ramarag) [inline build.cmd]: done

**Comment by ramarag** [comment]
@dotnet-bot test OSX x64 Release Build and Test

**Comment by ramarag** [comment]
@dotnet-bot test OSX x64 Release Build and Test

**Comment by gkhanna79** [comment]
The test following test failed:

http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/release_osx_prtest/1583/testReport/junit/miscellaneous/queryperformancecounter/test1/

**Comment by ramarag** [comment]
@dotnet-bot test OSX x64 Release Build and Test

**Comment by jkotas** [comment]
This change completely killed incremental builds. I have opened #2486

**Pull request merged**